gateway/errors.go

```diff
@@ -153,7 +155,18 @@ func webError(w http.ResponseWriter, err error, defaultCode int) {
 		code = gwErr.StatusCode
 	}
 
-	http.Error(w, err.Error(), code)
+	acceptsHTML := strings.Contains(r.Header.Get("Accept"), "text/html")
+	if acceptsHTML {
+		w.WriteHeader(code)
+		_ = assets.ErrorTemplate.Execute(w, assets.ErrorTemplateData{
+			Path:       r.URL.Path,
```

Comment:
:warning:  accessing URL path directly here may cause bugs – if not today, then in the future, after refactors (it happened already a while ago). 

Rule of thumb for `boxo/gateway` is to avoid `r.URL.Path` and only acting on already parsed `contentPath` -- this way there is no room for bugs caused by content path normalization (now or in future).

So.. we need to pass `contentPath ipath.Path` to  `webError` and use `contentPath.String()` here, just like we do in the dir listing template  in `handler_unixfs_dir.go`

Reply:
We actually don't need it for this page. Removed. Besides, `webError` is called many times before `contentPath` even exists. If we want to pass it, we have to pass it around in many places we don't right now and see what to do when it doesn't exist yet. But I'll remove for now as we don't use it on the error page.